File: frontend/src/types/fleet.ts
// src/types/fleet.ts

export interface ErrorInfo {
    timestamp: string;
    type: string;
    description: string;
    severity: string; // "WARNING" or "FATAL"
  }
  
  export interface AGVInfo {
    serial: string;
    manufacturer: string;
    connection: string;
    battery: number;
    operating_mode: string;
    position: [number, number]; // [longitude, latitude]
    theta: number;
    last_update: string;
    connect_timestamp: number;
    current_order: string | null;
    errors: ErrorInfo[];
    sensor_status: Record<string, string> | null;
  }
  
  export type FleetDict = Record<string, AGVInfo>;

File: backend/app/models/fleet.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import List, Optional, Tuple, Dict
import time

class ErrorInfo(BaseModel):
    timestamp: datetime
    type: str
    description: str
    severity: str # "WARNING" or "FATAL"

class AGVInfo(BaseModel):
    serial: str
    manufacturer: str
    connection: str = "OFFLINE"
    battery: float = 0.0
    operating_mode: str = ""
    # Store as (longitude, latitude)
    position: Tuple[float, float] = (0.0, 0.0) 
    theta: float = 0.0
    last_update: datetime = Field(default_factory=datetime.utcnow)
    connect_timestamp: float = Field(default_factory=time.time)
    current_order: Optional[str] = None
    errors: List[ErrorInfo] = []
    # e.g., {"IMU": "OK", "Laser": "WARN"}
    sensor_status: Optional[Dict[str, str]] = None